src/components/LineUp/index.tsx

```diff
@@ -0,0 +1,495 @@
+import { useState } from "react"
+import styled from "styled-components"
+import { Artist, schedule } from "./data"
+import rightDiamond from "../../img/music-festival/right-diamond.png"
+// import rightDiamond from "../img/music-festival/right-diamond.png"
+import { Modal } from "../Modal/Index"
+import { Title } from "../Title"
+
+enum DayHighlight {
+  FULL = "FULL",
+  ONE = 1,
+  TWO = 2,
+  THREE = 3,
+  FOUR = 4,
+}
+
+function LineUp() {
+  const [highlightedDay, setHighlightedDay] = useState(DayHighlight.FULL)
+  const [modalOpen, setModalOpen] = useState(false)
+  const [selectedArtist, setSelectedArtist] = useState<Artist | null>(null)
+
+  const higlightDay = (day: DayHighlight): void => {
+    setHighlightedDay(day)
+  }
+
+  const isHighlighted = (index: number): boolean => {
+    return highlightedDay === index + 1
+  }
+
+  const handleArtistClick = (artist: Artist) => {
+    setSelectedArtist(artist)
+    setModalOpen(true)
+  }
+
+  return (
+    <SectionWrapper id="lineup">
+      <Title>Line Up</Title>
+      <DaysContainer>
+        <FullLineUp
+          onClick={() => higlightDay(DayHighlight.FULL)}
+          gradientTo={dayColors[1]}
+          color={dayColors[0]}
+        >
+          <span>FULL</span>
+          <span>LINE-UP</span>
+        </FullLineUp>
+        <Day
+          color={dayColors[0]}
+          onClick={() => higlightDay(DayHighlight.ONE)}
+          gradientTo={dayColors[1]}
+          isActive={highlightedDay === DayHighlight.ONE}
+          className="nasalization"
+        >
+          <span>DAY ONE</span>
+          <span>Nov. 20</span>
+        </Day>
+        <Day
+          color={dayColors[1]}
+          onClick={() => higlightDay(DayHighlight.TWO)}
+          gradientTo={dayColors[2]}
+          isActive={highlightedDay === DayHighlight.TWO}
+        >
+          <span>DAY TWO</span>
+          <span>Nov. 21</span>
+        </Day>
+        <Day
+          color={dayColors[2]}
+          onClick={() => higlightDay(DayHighlight.THREE)}
+          gradientTo={dayColors[3]}
+          isActive={highlightedDay === DayHighlight.THREE}
+        >
+          <span>DAY THREE</span>
+          <span>Nov. 22</span>
+        </Day>
+        <Day
+          color={dayColors[3]}
+          onClick={() => higlightDay(DayHighlight.FOUR)}
+          gradientTo={dayColors[3]}
+          isActive={highlightedDay === DayHighlight.FOUR}
+        >
+          <span>DAY FOUR</span>
+          <span>Nov. 23</span>
+        </Day>
+      </DaysContainer>
+      <Container>
+        <BackgroundImage />
+        <Content>
+          <Timeline columns={timeSlots.length}>
+            <TimelineLine />
+            {timeSlots.map((time) => (
+              <TimeSlot key={time}>
+                <TimeDiamond src={rightDiamond} alt="diamond" />
+                <TimeText>{time} UTC</TimeText>
+                <TimeText>{parseInt(time) - 7}:00 PST</TimeText>
```

Comment:
when `time >= 0 && time < 7`, the hour is negative

const convertToPST(int hour) => hour >= 7 ? (hour - 7) : ((hour + 24) - 7);